### PR TITLE
Bug fixes in main branch

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by tylerjw — APPROVED** [review]
Please update the description to describe what you are changing here.

**Review by tylerjw — APPROVED** [review]
new changes look good

File: moveit_core/robot_trajectory/src/robot_trajectory.cpp [hunk 1]
```diff
@@ -258,27 +255,24 @@ void RobotTrajectory::getRobotTrajectoryMsg(moveit_msgs::msg::RobotTrajectory& t
   if (!onedof.empty())
   {
     trajectory.joint_trajectory.header.frame_id = robot_model_->getModelFrame();
-    trajectory.joint_trajectory.header.stamp = stamp;
+    trajectory.joint_trajectory.header.stamp = rclcpp::Time(0, 0, RCL_ROS_TIME);
```
Comment (AndyZe) [inline moveit_core/robot_trajectory/src/robot_trajectory.cpp:258]: Looking at the documentation for rclcpp::Time, it looks like the default is RCL_SYSTEM_TIME. What's the reason for something different here?

File: moveit_core/robot_trajectory/src/robot_trajectory.cpp [hunk 2]
```diff
@@ -258,27 +255,24 @@ void RobotTrajectory::getRobotTrajectoryMsg(moveit_msgs::msg::RobotTrajectory& t
...
     trajectory.joint_trajectory.points.resize(waypoints_.size());
   }
 
   if (!mdof.empty())
   {
     trajectory.multi_dof_joint_trajectory.header.frame_id = robot_model_->getModelFrame();
-    trajectory.multi_dof_joint_trajectory.header.stamp = stamp;
+    trajectory.multi_dof_joint_trajectory.header.stamp = rclcpp::Time(0, 0, RCL_ROS_TIME);
```
Comment (AndyZe) [inline moveit_core/robot_trajectory/src/robot_trajectory.cpp:265]: Same question

**Review by AndyZe — COMMENTED** [review]
Just a question, then +1

File: moveit_core/robot_trajectory/src/robot_trajectory.cpp (hunk 1 above)
Comment (henningkayser) [reply moveit_core/robot_trajectory/src/robot_trajectory.cpp:258]: RCL_ROS_TIME is default for node time as it should be able to support simulation and is mostly used for messages.

**Review by henningkayser — APPROVED** [review]

**Pull request merged**